packages/autocomplete-plugin-recent-searches/src/usecases/localStorage/search.ts

```diff
@@ -10,12 +34,13 @@ export function search<TItem extends RecentSearchesItem>({
   query,
   items,
   limit,
-}: SearchParams<TItem>) {
+}: SearchParams<TItem>): Array<Highlighted<TItem>> {
   if (!query) {
-    return items.slice(0, limit);
+    return items.slice(0, limit).map((item) => highlight({ item, query }));
   }
 
   return items
-    .filter((item) => item.query.toLowerCase().startsWith(query.toLowerCase()))
-    .slice(0, limit);
+    .filter((item) => item.query.toLowerCase().includes(query.toLowerCase()))
```

Comment:
if you use `indexOf` here (instead of includes), you already have the start index of the highlight & the end index will be that + the length of the query. You'd need to store this info temporarily on the object, and only highlight matches though, which will be slightly more efficient